lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -2882,8 +2888,8 @@ def main():
             usage_msg='set publish.mode'
         ),
         healthcheck_start_period=dict(
-            docker_py_version='2.4.0',
-            docker_api_version='1.25',
+            docker_py_version='2.6.0',
```

Comment:
Are you sure about the 2.6.0 change? According to the changelog (https://docker-py.readthedocs.io/en/stable/change-log.html) for 2.4.0, `Added support for start_period configuration in docker.types.Healthcheck.`.

Reply:
Hmm yes that looks wrong. I changed it to align with the main `healthcheck` param which is marked as docker-py 2.6.0 https://github.com/ansible/ansible/blob/77f3babc7d0da631a73dd8a72c456d10d8098ef0/lib/ansible/modules/cloud/docker/docker_swarm_service.py#L2868

I need to look into if that’s not the right version.

Reply:
Ah, ok. In that case 2.6.0 seems to be ok; the 2.6.0 changelog actually mentions: `Added support for the following parameters when creating a ContainerSpec object: groups, open_stdin, read_only, stop_signal, helathcheck, hosts, ns_config, configs, privileges.`

I guess `docker.types.Healthcheck` can only be used with regular containers in 2.4.0 then.

Reply:
> I guess docker.types.Healthcheck can only be used with regular containers in 2.4.0 then.

Yes that must be the case.